llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2242,6 +2246,19 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
     }
   }
 
+  VPValue *HeaderMask = findHeaderMask(Plan);
+  if (!HeaderMask) {
+    // TODO: Remove this once we move transformRecipesToEVLRecipes into
+    // tryToBuildVPlanWithVPRecipes.
+    for (VPRecipeBase *R : reverse(ToErase)) {
+      SmallVector<VPValue *> PossiblyDead(R->operands());
+      R->eraseFromParent();
+      for (VPValue *Op : PossiblyDead)
+        recursivelyDeleteDeadRecipes(Op);
+    }
+    return;
+  }
```

Comment:
Once we move the variable step transform into tryToBuildVPlanWithVPRecipes then we can just rely on the removeDeadRecipes pass.